sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt

```diff
@@ -1,64 +1,61 @@
  */
-internal class TrackRequest (
-        private var event: String,
-        private var customerProperties: KlaviyoCustomerProperties,
-        private var properties: KlaviyoEventProperties? = null
-): KlaviyoRequest() {
+internal class TrackRequest(
+    apiKey: String,
+    event: KlaviyoEvent,
+    customerProperties: KlaviyoCustomerProperties,
+    properties: KlaviyoEventProperties? = null
+) : KlaviyoRequest() {
     internal companion object {
-        const val TRACK_ENDPOINT = "api/track"
+        const val TRACK_ENDPOINT = "client/events"
     }
 
-    private var timestamp: Long? = null
-
     override var urlString = "$BASE_URL/$TRACK_ENDPOINT"
-    override var requestMethod = RequestMethod.GET
+    override var requestMethod = RequestMethod.POST
 
-    /**
-     * Builds a JSON payload suitable for a track request and returns it as a String
-     * Appends external information to the customer properties map before serializing it to JSON
-     *
-     * For more information on the structure of Klaviyo requests please reference the API docs:
-     * https://www.klaviyo.com/docs
-     *
-     * @return JSON payload as a string
-     */
-    override fun buildKlaviyoJsonQuery(): String {
-        customerProperties.setAnonymousId()
-        
-        val json = JSONObject(
-            mapOf(
-                "token" to KlaviyoConfig.apiKey,
-                "event" to event,
-                "customer_properties" to JSONObject(customerProperties.toMap()),
-                "properties" to properties?.let { JSONObject(properties?.toMap()) },
-                "time" to timestamp?.let { it }
-            ).filterValues { it != null }
-        )
+    override var queryData: Map<String, String> = mapOf(
+        "company_id" to apiKey
+    )
 
-        return json.toString()
-    }
+    override var payload: String? =
+        JSONObject(
+            mapOf(
+                "data" to mapOf(
+                    "type" to "event",
+                    "attributes" to mapOf(
+                        "metric" to mapOf(
+                            "name" to event.name,
+                        ),
+                        "profile" to JSONObject(customerProperties.setAnonymousId().toMap()),
+                        "properties" to properties?.let { JSONObject(it.toMap()) },
+                        "time" to timestamp,
+                    ).filterValues { it != null }
+                )
+            )
+        ).toString()
 
-    /**
-     * Generates a Unix timestamp to store as the [timestamp] on this object
-     */
-    internal fun generateUnixTimestamp() {
-        timestamp = System.currentTimeMillis() / 1000L
+    override fun appendHeaders(connection: HttpURLConnection) {
+        connection.setRequestProperty("Content-Type", "application/json")
+        connection.setRequestProperty("Accept", "application/json")
+        connection.setRequestProperty("Revision", "2022-10-17")
```

Comment:
Should this revision number by a constant somewhere?

Reply:
possibly but small change can be address after the APIs are more solid

Reply:
Yah maybe not a bad idea but I doubt it'll be referenced anywhere else.

Reply:
Totally fair ya

Reply:
Actually if you aren't using v3 you don't need these at all.

Reply:
But probably harmless.

Reply:
track is using v3

Reply:
identify is still on v1

Reply:
Nice!